### PR TITLE
Docs shervin

### PR DESCRIPTION
**Pull request by shervin86**
- I added type hints in several places, not everywhere yet. 
- I added the sphinx extension to pick the type hints for the return type in the documentation
- I changed some of the initialization defaults for internal string members from None to an empty string whenever a None value was not used anywhere else in the code.
- I added some metadata in the `__init__.py` file like authors, version number, etc. that are then read by sphinx when creating the documentation. This avoid writing the same information in several places.
- Reformatted part of the code with blank.

### CODE REVIEW

File: libpyvinyl/BaseCalculator.py [hunk 1]
```diff
@@ -56,16 +60,16 @@ class BaseCalculator(AbstractBaseClass):
...
-    `test/integration/plusminus`
+    ``test/integration/plusminus``
 
     """
 
     def __init__(
         self,
         name: str,
-        input: Union[DataCollection, list, BaseData],
+        input: Union[DataCollection, List[DataCollection], BaseData],
```
Comment (JunCEEE) [inline libpyvinyl/BaseCalculator.py:70]: input cannot be a list of `DataCollection` while the return of the iterator has to be a `BaseData`.
```suggestion
        input: Union[DataCollection, List[DataCollection], BaseData],
```

File: libpyvinyl/BaseCalculator.py [hunk 2]
```diff
@@ -74,47 +78,40 @@ def __init__(
...
+                                    exact output file path depends on `instrument_base_dir` and
+                                    `calculator_base_dir`: `instrument_base_dir`/`calculator_base_dir`/filename
 
         :param parameters: The parameters for this calculator.
-        :type  parameters: Parameters
 
         """
         # Initialize the variables
-        self.__name = None
+        self.__name = ""
```
Comment (JunCEEE) [inline libpyvinyl/BaseCalculator.py:106]: I don't know if it's the standard way to initialize a variable with an empty string. Maybe @CFGrote can comment on that?

File: README.md [hunk 3]
```diff
@@ -5,9 +5,48 @@
...
+Simulations provide invaluable insights to plan (before) and
+ understand (after) experiments at neutron and X-ray facilities. 
+A wide set of libraries and programs are already available to simulate
+ neutron and X-ray beams, propagate them through the beamlines,
+ interact with matter of a sample and get data acquired by detectors. 
+
+The aim of this library is to provide a high level API to allow users
+ to access simulations with a unified interface, reaching almost
+ seamless interoperability, allowing to chain multiple
+ simulation softwares in a natural way.
```
Comment (CFGrote) [inline README.md:17]: The aim of  *libpyvinyl* is to provide a high level neutron and x-ray simulation API. With this harmonized user interface we achieve seamless interoperability of individual simulations thereby facilitating the concatenation of simulation steps into a simulation pipeline.

File: README.md [hunk 4]
```diff
@@ -5,9 +5,48 @@
...
+The vast differences with respect to parameter
+ names, unit conventions, configuration syntax, i.e. the user
+ interface, is, hence, overcome creating a `libpyvinyl` complaiant API
+ for each simulation software. 
+
+## Software specific APIs based on libpyvinyl
+The python package `libpyvinyl` provides a way to harmonize the user interfaces of such simulation codes. It is an object oriented library; its classes define the user interface to simulation codes, simulation parameters and simulation data.
 
+For a given simulation code, e.g. propagation of neutron or photon beams through a beamline, a new class would have to be defined that derives from the
+`libpyvinyl` classes. Implementing the methods meant to configure a simulation, launch the simulation code, and retrieve the output data. Since the
```
Comment (CFGrote) [inline README.md:27]: This  derived class requires the implementation of certain methods meant to ...

File: README.md [hunk 5]
```diff
@@ -5,9 +5,48 @@
...
+interplay between parametrization, execution, and IO is already taken care of at the level of `libpyvinyl`'s base classes, the effort to define a specialized interface (parameters, backengine and DataClass) for a new simulation code is rather minimal. This structure allows integrating simulation codes into simulation pipelines in the above sense.
```
Comment (CFGrote) [inline README.md:28]: ~~DataClass~~ data object

File: README.md [hunk 6]
```diff
@@ -5,9 +5,48 @@
...
+Three kind of users are the target of this package:
+ 1. developers of packages based on libpyvinyl offering new calculators
+   for simulations
+ 1. users wishing to run a simulation giving some inputs and retrieving
+   the results
+ 1. research facility experts that what to implement detailed
+   simulation of existing instruments at their facility or  willing to
+   design new ones
+ 
 The fundamental class is the `BaseCalculator` and its sister class `Parameters`.
```
Comment (CFGrote) [inline README.md:50]: This paragraph is partly duplicating the above

**Review by CFGrote — CHANGES_REQUESTED** [review]
Thanks everyone for your efforts. Only a few typos and wording issues in the Overview.

File: README.md [hunk 7]
```diff
@@ -5,9 +5,48 @@
 
 ## Overview
 
-The python package `libpyvinyl` exposes the high level API for simulation codes under
-the umbrella of the Virtual Neutron and x-raY Laboratory (ViNYL).
+Simulations provide invaluable insights to plan (before) and
+ understand (after) experiments at neutron and X-ray facilities. 
+A wide set of libraries and programs are already available to simulate
+ neutron and X-ray beams, propagate them through the beamlines,
```
[the remark this answers is not in the document]
Comment (JunCEEE) [reply README.md:11]: Even though it's okay to spell it in lower case, the most common spelling is still "X-ray" in my readings in the community.
See:
https://en.wikipedia.org/wiki/X-ray
https://www.britannica.com/science/X-ray-source
https://www.nature.com/articles/35021099

[the remark this answers is not in the document]
Comment (shervin86) [reply README.md:11]: I don't have strong opinions. What matters is to be consistent.

File: README.md [hunk 8]
```diff
@@ -5,9 +5,48 @@
...
+ interact with matter of a sample and get data acquired by detectors. 
+
+The aim of this library is to provide a high level API to allow users
+ to access simulations with a unified interface, reaching almost
+ seamless interoperability, allowing to chain multiple
+ simulation softwares in a natural way. 
+The vast differences with respect to parameter
+ names, unit conventions, configuration syntax, i.e. the user
+ interface, is, hence, overcome creating a `libpyvinyl` complaiant API
```
[the remark this answers is not in the document]
Comment (shervin86) [reply README.md:20]:
```suggestion
 interface, is, hence, overcome creating a `libpyvinyl` compliant API
```

File: README.md (hunk 3 above)
Comment (shervin86) [reply README.md:17]: ok

File: README.md (hunk 4 above)
Comment (shervin86) [reply README.md:27]: ok

**Comment by JunCEEE** [comment]
I don't support removing `Installation` from `README.md`. I consider it essential in `README.md`, even though `libpyvinyl` is designed to be used by other package developers.

File: README.md [hunk 9]
```diff
@@ -30,109 +78,16 @@ how to declare a derived `Calculator` and implement a `backengine` method. The e
...
 ## libpyvinyl projects
-- McStatsScript
-- SimEx-Lite
-
+There are currently two projects based on libpyvinyl:
+- McStatsScript: [https://github.com/PaNOSC-ViNYL/McStasScript](https://github.com/PaNOSC-ViNYL/McStasScript)
+- SimEx-Lite:[https://github.com/PaNOSC-ViNYL/SimEx-Lite](https://github.com/PaNOSC-ViNYL/SimEx-Lite)
 
+		
 ## Overview of base classes
```
Comment (JunCEEE) [inline README.md:88]: This section is not needed in README.md. It's a part of the user guide.

File: README.md [hunk 10]
```diff
@@ -30,109 +78,16 @@ how to declare a derived `Calculator` and implement a `backengine` method. The e
...
+![libpyvinyl](libpyvinyl.drawio.svg)
+(Edit)[shorturl.at/nowBK]
```
Comment (JunCEEE) [inline README.md:90]: This section is not needed in README.md. It's a part of the user guide.

File: libpyvinyl/BaseCalculator.py (hunk 1 above)
Comment (shervin86) [reply libpyvinyl/BaseCalculator.py:70]: Understood

File: libpyvinyl/BaseCalculator.py (hunk 2 above)
Comment (shervin86) [reply libpyvinyl/BaseCalculator.py:106]: Ok

**Pull request merged**